From 659e2c37862e4cfc3a5a76ac3bc250171514c1d6 Mon Sep 17 00:00:00 2001
From: Wally Gleemer <Wally.Gleemer@windriver.com>
Date: Tue, 25 Mar 2008 16:21:37 -0700
Subject: [PATCH] cavium: get rid of warnings from i8259 code

Capture the request_irq return value so that we don't get nagged
by gcc.

Signed-off-by: Wally Gleemer <Wally.Gleemer@windriver.com>
---
 arch/mips/cavium-octeon/i8259.c |    6 +++---
 1 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/arch/mips/cavium-octeon/i8259.c b/arch/mips/cavium-octeon/i8259.c
index a251771..a903c56 100644
--- a/arch/mips/cavium-octeon/i8259.c
+++ b/arch/mips/cavium-octeon/i8259.c
@@ -125,7 +125,7 @@ static irqreturn_t octeon_i8259_interrupt(int cpl, void *dev_id)
 
 void octeon_i8259_setup(int irq_line)
 {
-	printk("8259: Initializing\n");
+	int r;
 
 	/* Setup the Master 8259 */
 	outb(ICW1 | ICW1_NEED_ICW4, ICW1_ADDRESS);	/* Begin the init
@@ -165,11 +165,11 @@ void octeon_i8259_setup(int irq_line)
 		printk("8259: Setting GPIO %d for the interrupt\n",
 		       irq_line - 24);
 		cvmx_write_csr(CVMX_GPIO_BIT_CFGX(irq_line - 24), 0x114);
-		request_irq(irq_line, octeon_i8259_interrupt, SA_SHIRQ, "8259",
+		r = request_irq(irq_line, octeon_i8259_interrupt, IRQF_SHARED, "8259",
 			    octeon_i8259_interrupt);
 	} else if ((irq_line >= 44) && (irq_line <= 47)) {
 		printk("8259: Using PCI INT-%c\n", irq_line - 44 + 'A');
-		request_irq(irq_line, octeon_i8259_interrupt, SA_SHIRQ, "8259",
+		r = request_irq(irq_line, octeon_i8259_interrupt, IRQF_SHARED, "8259",
 			    octeon_i8259_interrupt);
 	} else {
 		panic("8259: Don't know how to setup the interrupt IRQ %d\n",
-- 
1.5.5.1

